libchisel/src/remapstart.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::HashMap;
+
+use super::{ModuleError, ModulePreset, ModuleTranslator};
+use parity_wasm::elements::*;
+
+pub struct RemapStart {}
+
+impl ModulePreset for RemapStart {
+    fn with_preset(preset: &str) -> Result<Self, ()> {
+        match preset {
+            "ewasm" => Ok(RemapStart {}),
+            _ => Err(()),
+        }
+    }
+}
+
+impl ModuleTranslator for RemapStart {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Ok(remap_start(module))
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let mut ret = module.clone();
+        let modified = remap_start(&mut ret);
+        if modified {
+            return Ok(Some(ret));
```

Comment:
can fold this into a cleaner expression without the c-style return statements like so:
```rust
if remap_start(&mut ret) {
    Ok(Some(ret))
} else {
    Ok(None)
}
```